### PR TITLE
License and Maven update

### LINKED ISSUE #241: Update license and maven configuration
Project needs and update of license mentions, maven version. update and maven configuration simplification as discussed in #239

### PR DESCRIPTION
**Pull request by antoinesd**
fixes #241
Update license to remove year and apply it to existing files
Update Maven version to 3.9.6
Update Maven dependencies and plugin to last non-breaking version

### CODE REVIEW

File: etc/license.txt [hunk 1]
```diff
@@ -1,13 +1,5 @@
- Copyright 2017 - 2023 The ModiTect authors
+ SPDX-License-Identifier: Apache-2.0
 
- Licensed under the Apache License, Version 2.0 (the "License");
- you may not use this file except in compliance with the License.
- You may obtain a copy of the License at
+  Copyright The original authors
```
Comment (aalmiray) [inline etc/license.txt:3]: This line and the next have one extra space of indentation. Please make it only one space in total and reformat all files once again. Thank you. 😅

**Review by aalmiray — CHANGES_REQUESTED** [review]

File: etc/license.txt (hunk 1 above)
Comment (antoinesd) [reply etc/license.txt]: It was on purpose of course: to check the hawk eye of the reviewer 😉. You passed the test!
Thanks Andres!

**Review by gunnarmorling — APPROVED** [review]
Nice one, LGTM. Any idea though what's wrong with that failing test?

**Comment by antoinesd** [comment]
> Any idea though what's wrong with that failing test?

Hello @gunnarmorling. If ou referring the 4 occurrences of
```
[ERROR] Error occurred during initialization of boot layer
[ERROR] java.lang.module.FindException: Module com.example not found
```
It seems to come from the [`@Before`](https://github.com/moditect/moditect/blob/da100986b992c6cef2ab3d390af6617cea95dece/core/src/test/java/org/moditect/test/AddModuleInfoTest.java#L56) method in the test which doing deletion in target folder :
`[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1`
I'll try to investigate.

For the record, the error was already there before this PR and the update of the surefire plugin. It was just less visible (only present in the surefire dump file)

**Comment by antoinesd** [comment]
@aalmiray @gunnarmorling Are you ok to merge this PR ? I need it to rebase my 2 other PR and fix 3rd example

**Comment by aalmiray** [comment]
@antoinesd what do we do with the failing test? In the past those errors did not fail the build, did they?

**Comment by antoinesd** [comment]
@aalmiray did you read my [comment](https://github.com/moditect/moditect/pull/242#issuecomment-2003041449) ? These error were here before. The only change is that this new version of surefire plugin show them explicitly in the console and not only in the file `target/surefire-report/2024-***-jvmRun1.dumpstream`. If it's very problematic we can revert to previous surefire plugin version but we'll have to deal with this error anyway.

**Comment by aalmiray** [comment]
Yes, I read the comment. And at first I thought it was just error messages being displayed in a different way. But now the build also breaks snd that is what concerns me. Yes, the code was broken before but the build continued (which is wrong of course). Now the build breaks as it should, so we have to find a way to fix it.

**Comment by antoinesd** [comment]
@aalmiray @gunnarmorling ok, got it! It was a misunderstanding (a bug between my chair and keyboard in fact)  because on my Mac the build was always failing at `Vert.x` example  due to  my antivirus preventing the usage of `log4j 2.8` that contains the log4shell vulnerability.
After disabling my antivirus I have the build working on master. Updating the maven version lot 3.9.6 breaks it. 
What I suggest is to merge this PR with #238 and enhance the resulting PR with fix for hibernate validator example as well.
After that maven should be updated with most plugins up to date and the build should work.the other solution is to move maven update in another PR with all the fix to existing example.
Wdyt?

**Comment by aalmiray** [comment]
@antoinesd Alright! Let's do that. I'll merge both PRs.

**Pull request merged**